test/kubernetes_unittest.cc

```diff
@@ -509,4 +515,186 @@ TEST_F(KubernetesTest, GetContainerMetadata) {
+  EXPECT_EQ("k8s_container.TestPodUid.TestContainerName0",
+            metadata_list[1].ids()[0]);
+  EXPECT_EQ("k8s_container.TestNamespace.TestPodName.TestContainerName0",
+            metadata_list[1].ids()[1]);
+  EXPECT_EQ(MonitoredResource("k8s_container", {
+    {"cluster_name", "TestClusterName"},
+    {"container_name", "TestContainerName0"},
+    {"location", "TestClusterLocation"},
+    {"namespace_name", "TestNamespace"},
+    {"pod_name", "TestPodName"},
+  }), metadata_list[1].resource());
+  EXPECT_FALSE(metadata_list[1].metadata().ignore);
+  EXPECT_EQ("1.6", metadata_list[1].metadata().version);
+  EXPECT_FALSE(metadata_list[1].metadata().is_deleted);
+  EXPECT_EQ(time::rfc3339::FromString("2018-03-03T01:23:45.678901234Z"),
+            metadata_list[1].metadata().created_at);
+  EXPECT_EQ(Timestamp(), metadata_list[1].metadata().collected_at);
+  json::value container1 = json::object({
+    {"blobs", json::object({
+      {"association", json::object({
+        {"raw", json::object({
+          {"controllers", json::object({
+            {"topLevelControllerName", json::string("TestPodName")},
+            {"topLevelControllerType", json::string("Pod")},
+          })},
+          {"infrastructureResource", json::object({
+            {"labels", json::object({
+              {"instance_id", json::string("TestID")},
+              {"zone", json::string("TestZone")},
+            })},
+            {"type", json::string("gce_instance")},
+          })},
+          {"nodeName", json::string("TestSpecNodeName")},
+        })},
+        {"version", json::string("TestVersion")},
+      })},
+      {"spec", json::object({
+        {"raw", json::object({
+          {"name", json::string("TestContainerName0")},
+        })},
+        {"version", json::string("1.6")},
+      })},
+      {"status", json::object({
+        {"raw", json::object({
+          {"name", json::string("TestContainerName0")},
+        })},
+        {"version", json::string("1.6")},
+      })},
+    })},
+  });
+  EXPECT_EQ(container1->ToString(), metadata_list[1].metadata().metadata->ToString());
+
+  EXPECT_EQ(2, metadata_list[2].ids().size());
+  EXPECT_EQ("k8s_pod.TestPodUid", metadata_list[2].ids()[0]);
+  EXPECT_EQ("k8s_pod.TestNamespace.TestPodName", metadata_list[2].ids()[1]);
+  EXPECT_EQ(MonitoredResource("k8s_pod", {
+      {"cluster_name", "TestClusterName"},
+      {"location", "TestClusterLocation"},
+      {"namespace_name", "TestNamespace"},
+      {"pod_name", "TestPodName"},
+  }), metadata_list[2].resource());
+  EXPECT_FALSE(metadata_list[2].metadata().ignore);
+  EXPECT_EQ("TestVersion", metadata_list[2].metadata().version);
+  EXPECT_FALSE(metadata_list[2].metadata().is_deleted);
+  EXPECT_EQ(time::rfc3339::FromString("2018-03-03T01:23:45.678901234Z"),
+            metadata_list[2].metadata().created_at);
+  EXPECT_EQ(Timestamp(), metadata_list[2].metadata().collected_at);
+  json::value container2 = json::object({
```

Comment:
Should this not be pod metadata?

Reply:
I actually meant that this looks like container metadata (with the "metadata", "spec", and "status" fields). Should this not be pod metadata (with one "api" field)?

Reply:
Hmm, well reading the code, it does seem to me that a pod should be played at the end of the metadata list.  Truthfully, I had a hard time following all the code paths closely, so its difficult for me to evaluate whether this is an actual bug.

Reply:
Oh, silly me. This *is* pod metadata. I miscounted the nesting depth. Never mind.